sdks/java/io/kinesis/src/main/java/org/apache/beam/sdk/io/kinesis/ShardReadersPool.java

```diff
@@ -137,15 +189,16 @@ void stop() {
 
   boolean allShardsUpToDate() {
     boolean shardsUpToDate = true;
-    for (ShardRecordsIterator shardRecordsIterator : shardIteratorsMap.values()) {
+    for (ShardRecordsIterator shardRecordsIterator : shardIteratorsMap.get().values()) {
       shardsUpToDate &= shardRecordsIterator.isUpToDate();
     }
     return shardsUpToDate;
   }
 
   KinesisReaderCheckpoint getCheckpointMark() {
-    return new KinesisReaderCheckpoint(transform(shardIteratorsMap.values(),
+    return new KinesisReaderCheckpoint(transform(shardIteratorsMap.get().values(),
```

Comment:
Same here (and I'm more worried about the checkpoint correctness)

Reply:
And the same here, I've extracted the shardIteratorsMap before the loop to use the consistent snapshot